plotting_refactor/DataViewer.py

```diff
@@ -0,0 +1,172 @@
+from PySide6 import QtWidgets
+
+from DataViewerUI import Ui_DataViewer
+from PlotWidget import PlotWidget
+from DataTreeWidget import DataTreeWidget
+from PlotTreeWidget import PlotTreeWidget
+from DataCollector import DataCollector
+from DataTreeItems import PlotPageItem, DataItem
+from PlotTreeItems import TabItem, SubTabItem, PlotItem, PlottableItem
+from PlotModifiers import ModifierLinecolor, ModifierLinestyle, ModifierColormap, PlotModifier
+
+
+class DataViewer(QtWidgets.QWidget, Ui_DataViewer):
+    def __init__(self, main_window):
+        super(DataViewer, self).__init__()
+        self.setupUi(self)
+
+        self.main_window = main_window
+        self.datacollector = DataCollector()
+
+        self.dataTreeWidget = DataTreeWidget(self, self.datacollector)
+        self.plotTreeWidget = PlotTreeWidget(self)
+
+        self.cmdClose.clicked.connect(self.onShowDataViewer)
+        self.cmdAddModifier.clicked.connect(self.onAddModifier)
+        self.plotTreeWidget.dropSignal.connect(self.redrawAll)
+
+        self.setupMofifierCombobox()
+        self.plot_widget = PlotWidget(self.datacollector)
+        self.data_origin_fitpage_index = None
+
+    def create_plot(self, fitpage_index):
+        self.update_plot_tree(fitpage_index)
+        self.plot_widget.show()
+        self.plot_widget.activateWindow()
+
+    def update_datasets_from_collector(self):
+        # block signals to prevent currentItemChanged to be called. otherwise the program crashes, because it tries
+        # to access the current item.
+        self.dataTreeWidget.blockSignals(True)
+        self.dataTreeWidget.clear()
+        self.dataTreeWidget.blockSignals(False)
+
+        datasets = self.datacollector.get_datasets()
+        for i in range(len(datasets)):
```

Comment:
` for i, dataset in enumerate(datasets):` - This will give you both the index and indexed dataset. Calling `dataset[i]` has a time cost everytime it is invoked.